url0067cdd33d_goldolphin_Mi_tgz-pJ8-mi_regex_CommonJ8/src/mi/task/Waiter.java

```diff
@@ -0,0 +1,57 @@
+package mi.task;
+
+/**
+ * A task to wait the result of its antecedent.
+ * @param <TResult> result type.
+ * @author goldolphin
+ *         2014-09-06 18:05
+ */
+public class Waiter<TResult> extends SeqTask<TResult, TResult> {
+    private final Object lock = new Object();
+    private volatile boolean isComplete = false;
+    private volatile TResult result;
+
+    public Waiter(ITask<TResult> antecedent) {
+        super(antecedent, false);
+    }
+
+    /**
+     * Whether the task is complete.
+     * @return
+     */
+    public boolean isComplete() {
+        return isComplete;
+    }
+
+    /**
+     * Get the result in a blocking way.
+     * @return
+     */
+    public TResult getResult() {
+        synchronized (lock) {
+            while (!isComplete) {
+                try {
+                    lock.wait();
+                } catch (InterruptedException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)